pdfPreviewInMsg/background.js

```diff
@@ -0,0 +1,166 @@
+/*
+----------------------------------------------------------------
+This extension for Thunderbird was made by : e-gaulue & rholeczy
+                        © 2021
+----------------------------------------------------------------
+*/
+var initialized;
+
+
+/**
+ * Preview a pdf in the mail.
+ * 
+ * @param {*} pdf 
+ * @param {*} canvas 
+ * @param {*} nbP 
+ * @param {*} tabid 
+ * @param {*} filename 
+ * @param {*} messageid 
+ */
+async function pdfPreviewRH(pdf, canvas, nbP, tabid, filename, messageid) {
+  let page = await pdf.getPage(nbP);
+  let viewport = await page.getViewport({ scale: 1.0 });
+  let scale = Math.min(
+    canvas.width / viewport.width,
+    canvas.height / viewport.height
+  );
+  let thumbViewPort = await page.getViewport({ scale });
+  let context = canvas.getContext("2d");
+  let task = page.render({ canvasContext: context, viewport: thumbViewPort });
+  task.promise.then(() => {
+    // Convert the pdf's page to an image.
+    let urlImage = canvas.toDataURL("image/jpeg"); 
+    // Send a message to the tab to do ???
+    browser.tabs.sendMessage(tabid, {
+      type: "handle_me",
+      image: urlImage,
+      part_name: filename,
+      numPage: nbP,
+      messageId: messageid,
+    });
```

Comment:
What is happening here? What is send to the tab, and what is the tab doing with it?